ts/src/okx.ts

```diff
@@ -2146,17 +2146,14 @@ export default class okx extends Exchange {
         let defaultType = 'Candles';
         if (since !== undefined) {
             const now = this.milliseconds ();
-            const difference = now - since;
             const durationInMilliseconds = duration * 1000;
-            // if the since timestamp is more than limit candles back in the past
-            // additional one bar for max offset to round the current day to UTC
-            const calc = (1440 - limit - 1) * durationInMilliseconds;
-            if (difference > calc) {
+            // switch to history candles if since is past the cutoff for current candles
+            const historyBorder = now - (1440 * durationInMilliseconds);
```

Comment:
i am not sure whether it's true what I am saying, but for safety and guaranteed behavior, i would have done this:

```suggestion
            const historyBorder = now - (1440 - 1) * durationInMilliseconds);
```

the reason is that the current bar might fall under 1400th bar, so, it might be better to switch to `HistoryCandles` 1 bar earlier than exactly 1400th bar. 1 bar wouldn't much make any difference among users, however, it might save from unforeseen issues which might exist in exchange api because of current bar (which might be considered by API engine as 1401th bar) formation.

Reply:
Okay I think that makes sense I'll add that change

Reply:
I also think -1 is necessary, because after we substract 1439 full periods 1 last nonclosed candle period left. 1440 total

Reply:
But if okx exchange returns live Candles for full 1440 candles plus 1 non-closed, than -1 is not necessary. All depends on exchange side

Reply:
@Alfavisioner yes, it was just added as a "possible" safety measure, which might arise (however, unconfirmed). anayw, 1 bar is not any significant, however, worth for that safety check